x-pack/solutions/security/plugins/security_solution/public/data_view_picker/hooks/use_selected_patterns.ts

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import type { DataViewPickerScopeName } from '../constants';
+import { useDataView } from './use_data_view';
+
+export const useSelectedPatterns = (scope: DataViewPickerScopeName): string[] => {
```

Comment:
<3 Thank you for separating these out. I do wonder if we should use selectors for these for the caching benefit..